services/wordpressService.ts

```diff
@@ -0,0 +1,114 @@
+import { Observer, ObserverCallback, ObserverDestructor } from "../../core/Observer";
+import { LogService } from "../../core/LogService";
+import { WordpressClient } from "../../core/wordpress/WordpressClient"
+import { isWordpressPageDTO, WordpressPageDTO } from "../../core/wordpress/dto/WordpressPageDTO"
+import { isWordpressPagesDTO, WordpressPageListDTO } from "../../core/wordpress/dto/WordpressPageListDTO"
+import { HgFrontend } from "../HgFrontend";
+import { WordpressReferenceDTO } from "../../core/wordpress/dto/WordpressReferenceDTO";
+import { WordpressUserProfileDTO } from "../../core/wordpress/dto/WordpressUserProfileDTO";
+
+HgFrontend.initialize(); // FIXME: Wasn't certain where this should be placed, but Im certain it needs to be fixed!
+
+export enum WordpressServiceEvent {
+    WORDPRESS_PAGE_ADDED = "WordpressService:WordpressPageAdded",
+    WORDPRESS_PAGE_REMOVED = "WordpressService:WordpressPageRemoved",
+    WORDPRESS_PAGE_UPDATED = "WordpressService:WordpressPageUpdated",
+    WORDPRESS_PAGE_CHANGED = "WordpressService:WordpressPageChanged"
+}
+
+export type WordpressServiceDestructor = ObserverDestructor;
+
+const LOG = LogService.createLogger('WordpressService');
+
+export class WordpressService {
+
+    private static _wordpressPage: WordpressPageDTO | undefined;
+    private static _wordpressReferences: WordpressReferenceDTO | undefined;
+    private static _observer: Observer<WordpressServiceEvent> = new Observer<WordpressServiceEvent>("WordpressService");
+
+    public static getCurrentWordpressPageId(): string | undefined {
```

Comment:
I think this service should be converted as a standard class and take the URL from the constructor. That way the service can be used multiple times for different wordpress URLs in the same app. Of course it could still have a static version, but the URL must be configurable. This library shouldn't even mention our website CMS URL.

Reply:
Added constructor that takes url and endpoint as parameters.